google/cloud/pubsub_v1/subscriber/message.py

```diff
@@ -281,16 +317,70 @@ def modify_ack_deadline(self, seconds: int) -> None:
                 against.
         """
         self._request_queue.put(
-            requests.ModAckRequest(ack_id=self._ack_id, seconds=seconds)
+            requests.ModAckRequest(ack_id=self._ack_id, seconds=seconds, future=None)
         )
 
+    def modify_ack_deadline_with_response(self, seconds: int) -> "pubsub_v1.subscriber.futures.Future":
+        """Resets the deadline for acknowledgement and returns the response
+        status via a future.
```

Comment:
Same here, we can de-ducplicate the code by introducing an optional `with_response` kwarg.